### ISSUE TITLE
Print orientation

### ISSUE THREAD

**Issue by amirulasrof** [andrey-ushakov/esc_pos_printer#23]
Hi @andrey-ushakov thank you a lot for your hardwork. Is it possible to print content in landscape view.
Refer sample below :

![0cc0a340-f22e-11e6-9e03-e2438ca31ee7](https://user-images.githubusercontent.com/11022660/71142033-15b37800-2251-11ea-9942-ef3f4a6905bd.jpg)

**Comment by DavBfr** [comment]
@amirulasrof try with this: https://gist.github.com/DavBfr/bbe8ba482b1a9101628d8b0d83888517
If it works correctly, it should be easy to make a nice coupon using <https://pub.dev/packages/pdf>.

**Comment by amirulasrof** [comment]
Thank you @DavBfr  I will definitely try this. Looks promising.

**Comment by amirulasrof** [comment]
Hi @DavBfr I try your code and encountered several error. 

This is my pubspec : [https://gist.github.com/amirulasrof/655099e425e559b70a8c3c344c956952](url)

Below is the error : 

Compiler message:
lib/main.dart:71:1: Error: This requires the 'extension-methods' experiment to be enabled.
Try enabling this experiment by adding it to the command line when compiling and running.
extension PdfRasterExt on PdfRaster {
^^^^^^^^^
lib/main.dart:71:27: Error: A function declaration needs an explicit list of parameters.
Try adding a parameter list to the function declaration.
extension PdfRasterExt on PdfRaster {
                          ^^^^^^^^^
lib/main.dart:71:1: Error: Type 'extension' not found.
extension PdfRasterExt on PdfRaster {
^^^^^^^^^
lib/main.dart:71:24: Error: Type 'on' not found.
extension PdfRasterExt on PdfRaster {
                       ^^
lib/main.dart:71:1: Error: 'extension' isn't a type.
extension PdfRasterExt on PdfRaster {
^^^^^^^^^
lib/main.dart:73:31: Error: Getter not found: 'width'.
    return im.Image.fromBytes(width, height, pixels);
                              ^^^^^
lib/main.dart:73:38: Error: Getter not found: 'height'.
    return im.Image.fromBytes(width, height, pixels);
                                     ^^^^^^
lib/main.dart:73:46: Error: Getter not found: 'pixels'.
    return im.Image.fromBytes(width, height, pixels);
                                             ^^^^^^
lib/main.dart:171:37: Error: Method not found: 'Printing.raster'.
    await for (var page in Printing.raster(_buildDocument(), dpi: 300)) {
                                    ^^^^^^
lib/main.dart:172:18: Error: Method not found: 'PdfRasterImage'.
      images.add(PdfRasterImage(page));
                 ^^^^^^^^^^^^^^
lib/main.dart:172:18: Error: The method 'PdfRasterImage' isn't defined for the class '_MyAppState'.
 - '_MyAppState' is from 'package:scsgoprint/main.dart' ('lib/main.dart').
Try correcting the name to the name of an existing method, or defining a method named 'PdfRasterImage'.
      images.add(PdfRasterImage(page));
                 ^^^^^^^^^^^^^^
lib/main.dart:180:37: Error: Method not found: 'Printing.raster'.
    await for (var page in Printing.raster(_buildDocument(), dpi: printerDpi)) {

**Comment by DavBfr** [comment]
In your pubspec, you missed:

```yaml
environment:
  sdk: ">=2.7.0 <3.0.0"
``` 

This code uses extension methods from Dart 2.7

**Comment by amirulasrof** [comment]
Its working now, thank you @DavBfr for the help.

**Comment by amirulasrof** [comment]
Hi @DavBfr how to implement barcode in the pdf. It is possible ?

**Comment by DavBfr** [comment]
Not yet, what type of barcode do you need?

**Comment by amirulasrof** [comment]
Any type will be good for now.

**Comment by DavBfr** [comment]
CODE 39 is implemented in pdf version 1.3.28:

```dart
BarcodeWidget(
  data: 'HELLO 123',
  width: 200,
  height: 50,
);
```

If you need to rotate it, wrap it with Transform.rotateBox:

```dart
Transform.rotateBox(
  angle: pi / 2,
  child: BarcodeWidget(
    data: 'HELLO 123',
    width: 200,
    height: 50,
  ),
);
```

**Comment by amirulasrof** [comment]
Very nice, thank you for the help @DavBfr
